docs/glossary.mdx

```diff
@@ -1,294 +1,376 @@
+keywords:
+  - zk
+  - zero knowledge
+  - what is mina
+  - mina ecosystem
+  - zkp
+  - mina
+  - zero-knowledge proofs
+---
+
 # Glossary
 
-### Account
+[A](#a) [B](#b) [C](#c) [D](#d) [E](#e) [F](#f) [I](#i) [J](#j) [K](#k) [L](#l) [M](#m) [N](#n) [O](#o) 
+[P](#p) [Q](#q) [R](#r) [S](#s) [T](#t) [U](#u) [V](#v) [Z](#z)
+
+Are you looking for a Mina term that isn't here yet? To let us know, please [create an issue](https://github.com/o1-labs/docs2/issues) or click **EDIT THIS PAGE** to submit a PR.
+
+## A
+
+### account
 
 Mina uses accounts to track each public key's state. This is distinct from Bitcoin's UTXO model of maintaining ledger state.
 
-### Archive Node
+### archive node
+
+A Mina node that stores the historical chain data to a persistent data source so it can later be retrieved. A zkApp can retrieve events and actions from one or more Mina [archive nodes](/node-operators/archive-node).
 
-Mina nodes are succinct by default, meaning they don't need to maintain historical information about the network, block, or transactions. An archive node stores the historical chain data to a persistent data source so it can later be retrieved.
+## B
 
-### Best Tip
+### best tip
 
-The best tip is the blockchain's latest block with the highest [chain strength](#chain-strength) known to the daemon.
+The blockchain's latest block with the highest [chain strength](#chain-strength) known to the daemon.
 
-### Block
+### block
 
-A set of transactions and consensus information that extend the state of the network. Includes a proof that the current state of the network is fully valid.
+A set of transactions and consensus information that extend the state of the network, a block includes a proof that the current state of the network is fully valid.
 
-### Blockchain
+### blockchain
 
 The data structure that is used in a cryptocurrency to maintain a shared state of all accounts in the network.
 
-### Block Confirmations
+### block confirmations
 
-The number of blocks added after the reference block. As the number of confirmations increases, the likelihood of a reorganization decreases, thereby increasing the likelihood of all transactions in the reference block being confirmed.
+The number of blocks added after the reference block. As the number of confirmations increases, the likelihood of a [reorganization](/glossary#reorganization) decreases, thereby increasing the likelihood of all transactions in the reference block being confirmed.
 
-### Block Fill Rate
+### block fill rate
 
-The proportion of [slots](#slot) that should contain a block. Some slots are intentionally empty to ensure the network can "catch up" in case of delay of messages.
+The proportion of [slots](#slot) that should contain a block. Some slots are intentionally empty to ensure the network can [catch up](/glossary#catchup) in case of delay of messages.
 
-### Block Producer
+### block producer
```

Comment:
Minor comment: Also, the people who run block producer nodes are referred to as as block producers.